airflow/contrib/hooks/jenkins_hook.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from airflow.hooks.base_hook import BaseHook
+
+import jenkins
+import distutils
+
+from airflow.utils.log.LoggingMixin import LoggingMixin
+
+class JenkinsHook(BaseHook):
+    """
+    Hook to manage connection to jenkins server
+    """
+
+    def __init__(self, conn_id='jenkins_default'):
+        connection = self.get_connection(conn_id)
+        self.connection = connection
+        connectionPrefix = 'http'
+        #connection.extra contains info about using https (true) or http (false)
+        if connection.extra == None or connection.extra == '':
+            connection.extra = 'false'
+            #set a default value to connection.extra to avoid rising ValueError in strtobool
+        if distutils.util.strtobool(connection.extra):
+            connectionPrefix = 'https'
+        url = '%s://%s:%d' % (connectionPrefix, connection.host, connection.port)
+        self.logger.info('Trying to connect to %s', url)
+        self.jenkins_server = jenkins.Jenkins(url, connection.login, connection._password)
```

Comment:
seems a typo here. should be connection.password not connection._password

Reply:
Fixed, both are working, but password seems more frequent in the code base so I switch to this one :). Good catch ;)